numpyro/distributions/distribution.py

```diff
@@ -386,3 +387,28 @@ def mean(self):
     @property
     def variance(self):
         raise NotImplementedError
+
+
+class Unit(Distribution):
+    """
+    Trivial nonnormalized distribution representing the unit type.
+
+    The unit type has a single value with no data, i.e. ``value.size == 0``.
+
+    This is used for :func:`numpyro.factor` statements.
+    """
+    arg_constraints = {'log_factor': real}
+    support = real
+
+    def __init__(self, log_factor, validate_args=None):
+        batch_shape = np.shape(log_factor)
+        event_shape = (0,)  # This satisfies .size == 0.
```

Comment:
Could you elaborate on why this 0 sized dim is needed?

Reply:
This is to make sure that the sample does not hold any data (only shape is available).